internal/tui/components/autocomplete/autocomplete.go

```diff
@@ -0,0 +1,270 @@
+	posY        int
+	width       int
+}
+
+type Config struct {
+	MaxVisible int
+	Width      int
+}
+
+func NewModel(ctx *context.ProgramContext) Model {
+	return Model{
+		ctx:        ctx,
+		visible:    false,
+		selected:   0,
+		maxVisible: 5,
+		width:      30,
+	}
+}
+
+func (m *Model) Configure(config Config) {
+	if config.MaxVisible > 0 {
+		m.maxVisible = config.MaxVisible
+	}
+	if config.Width > 0 {
+		m.width = config.Width
+	}
+}
+
+func (m Model) Update(msg tea.Msg) (Model, tea.Cmd) {
+	var cmd tea.Cmd
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch {
+		case key.Matches(msg, m.nextKey()):
+			m.Next()
+			return m, nil
+		case key.Matches(msg, m.prevKey()):
+			m.Prev()
+			return m, nil
+		}
+	}
+	return m, cmd
+}
+
+func (m *Model) nextKey() key.Binding {
+	return key.NewBinding(
+		key.WithKeys("down", "ctrl+n"),
+		key.WithHelp("↓/ctrl+n", "next suggestion"),
+	)
+}
+
+func (m *Model) prevKey() key.Binding {
+	return key.NewBinding(
+		key.WithKeys("up", "ctrl+p"),
+		key.WithHelp("↑/ctrl+p", "previous suggestion"),
+	)
+}
+
+func (m *Model) SetSuggestions(suggestions []string) {
+	m.suggestions = suggestions
+}
+
+func (m *Model) Filter(input string, excludeLabels []string) {
+	currentLabel := extractCurrentLabel(input)
+
+	excludeMap := make(map[string]bool)
+	for _, label := range excludeLabels {
+		excludeMap[strings.ToLower(strings.TrimSpace(label))] = true
+	}
+
+	m.filtered = filterAndRankSuggestions(
+		m.suggestions,
+		currentLabel,
+		excludeMap,
+		m.maxVisible,
+	)
+
+	m.selected = 0
+
+	m.visible = len(m.filtered) > 0 && strings.TrimSpace(currentLabel) != ""
+}
+
+func (m *Model) GetSelected() string {
+	if m.selected >= 0 && m.selected < len(m.filtered) {
+		return m.filtered[m.selected]
+	}
+	return ""
+}
+
+func (m *Model) Next() {
+	if len(m.filtered) > 0 {
+		m.selected = (m.selected + 1) % len(m.filtered)
+	}
+}
+
+func (m *Model) Prev() {
+	if len(m.filtered) > 0 {
+		m.selected--
+		if m.selected < 0 {
+			m.selected = len(m.filtered) - 1
+		}
+	}
+}
+
+func (m *Model) Show() {
+	m.visible = true
+}
+
+func (m *Model) Hide() {
+	m.visible = false
+}
+
+func (m *Model) IsVisible() bool {
+	return m.visible
+}
+
+func (m *Model) SetPosition(x, y int) {
+	m.posX = x
+	m.posY = y
+}
+
+func (m *Model) SetWidth(width int) {
+	m.width = width
+}
+
+func (m *Model) View() string {
+	if !m.visible || len(m.filtered) == 0 {
+		return ""
+	}
+
+	numVisible := min(len(m.filtered), m.maxVisible)
+
+	var b strings.Builder
+
+	maxLabelWidth := m.width - 4
+
+	for i := 0; i < numVisible && i < len(m.filtered); i++ {
+		label := m.filtered[i]
+		if len(label) > maxLabelWidth {
+			label = label[:maxLabelWidth-3] + "..."
+		}
+
+		// Style based on selection
+		if i == m.selected {
+			// Selected row - use inverted colors
+			selectedStyle := lipgloss.NewStyle().
+				Background(m.ctx.Theme.SelectedBackground).
+				Foreground(m.ctx.Theme.InvertedText)
+			b.WriteString(selectedStyle.Render("❯ " + label))
```

Comment:
maybe you can save the selection character in the constants file with the rest of the special characters.